### ISSUE TITLE
RxStorage API modification

### ISSUE THREAD

**Issue by brastak** [artipie/asto#159]
I have 2 proposals about RxStorage API:
1. change signature of list(Key) method: why we don't use Flowable<Key>? It fits better rx nature, in my opinion.
2. add error-free value() method. Currently we need to use exists().andThen(value()) approach to be sure that object exists in the storage. We can alter value() method to return Maybe<Content> (or we can add new method).

**Comment by 0crat** [comment]
@g4s8[/z](https://www.0crat.com/u/g4s8) please, pay attention to this issue

**Comment by g4s8** [comment]
@brastak I agree about 1 point. Also, I'd change `Storage.value` method to and return just `Publisher` without `CompletableFuture`.

But I don't really like 2 (error-free value), I think `RxStorage` should have same behavior as `Storage` to avoid confusion but using Rx API

**Comment by brastak** [comment]
@g4s8 Maybe we should change `Storage` API too? Getting value is our the most used operation, but now we have no way to execute it with one call. We should check key for existing firstly, and then retrieve. We provoke clients to use just `Storage.value` and handle exception (because of it should work almost 2 times faster) that does not look like good code design
